mikupad.html

```diff
@@ -2248,6 +2249,12 @@
 		if (elem.value === promptText) {
 			return;
 		} else if (elem.value.length && promptText.startsWith(elem.value)) {
+			const isTextSelected = elem.selectionStart !== elem.selectionEnd;
+			if (!isTextSelected && !preserveCursorPosition)
+			{
+				elem.value = promptText;
+				return;
+			}

```

Comment:
If you return like this the auto-scroll won't work, I guess this wasn't your intention.